apps/geo/tests/test_mutations.py

```diff
@@ -0,0 +1,48 @@
+from user.factories import UserFactory
+from project.factories import ProjectFactory
+from utils.graphene.tests import GraphQLTestCase
+
+
+class CreateTestMutation(GraphQLTestCase):
+    def test_create_region_in_project(self):
+        self.region_query = '''
+            mutation MyMutation($input: RegionInputType!){
+              createRegion(data: $input) {
+                ok
+                errors
+                result {
+                  id
+                  title
+                }
+              }
+            }
+        '''
+        user = UserFactory.create()
+        project = ProjectFactory.create(
+            title="dummy project 1",
+            created_by=user
+        )
+        project.add_member(user)
+
+        def _query_check(minput, **kwargs):
+            return self.query_check(
+                self.region_query,
+                minput=minput,
+                **kwargs
+            )
+    # -- Without login
+
+        minput = dict(
+            project=project.id,
+            code="NPL",
+            title="Test",
+        )
+        # without login
+        _query_check(minput, assert_for_error=True)
+
+        self.force_login(user)
+
+        _query_check(minput)
+        content = _query_check(minput)
+        self.assertEqual(content['data']['createRegion']['errors'], None)
+        self.assertEqual(content['data']['createRegion']['result']['title'], "Test")
```

Comment:
Let's add test for following

- User who is not member of project
- Different project that doesn't match region
 - published_region validation 
 - Created region is attached to project supplied.

Reply:
@sudan45 Let's handle all the cases mentioned above. 
For ordering, let's start from the low permission to high permission.

Reply:
Let's add more cases here

# Test
> NOTE: Add check to confirm there is no change in db even if there is permission error on response

## Region
- users
    - user (Which is used to create region)
    - other_user (Another normal user)
- Region
    - Region 01 (using user)
- Cases
    - Update Region (Not published)
        - other_user
            - Update region - Fail
            - Add admin - Fail
            - Update admin - Fail
        - user
            - Update region - Pass
            - Add admin - Pass
            - Update admin - Pass
    - Update Region (published)
        - other_user
            - Update region - Fail
            - Add admin - Fail
            - Update admin - Fail
        - user
            - Update region - Fail
            - Add admin - Fail
            - Update admin - Fail

## Region-Project
- users
    - non_member
    - normal_member
    - member (With access to modify Project)
- Region
    - published_region
    - pending_region
- Cases
    - No User - published_region - Fail
    - non_member - published_region - Fail
    - normal_member - published_region - Fail
    - member
        - pending_region - Fail
        - published_region - Pass